Makefile

```diff
@@ -857,9 +857,9 @@ THIRD_PARTY_SOURCES := $(wildcard $(SRC_DIR)/third-party/flatbuffers/*.cpp)
 HEADERS := $(wildcard $(SRC_DIR)/*.h)
 TESTSRC := $(wildcard tests/*.cpp)
 TESTHDR := $(wildcard tests/*.h)
-JSON_FORMATTER_SOURCES := $(wildcard tools/format/*.cpp) src/json.cpp
+JSON_FORMATTER_SOURCES := $(wildcard tools/format/*.cpp) src/cached_options.cpp src/json.cpp
```

Comment:
I don't feel like it's a good idea to introduce unrelated game constants into the formatter, which might make it more difficult to solve source dependency in the future and cause problems if someone accidentally uses an uninitialized constant.

Maybe it will be better to add `error_log_json_force_color` to `json.h/cpp` instead. (A better solution would of course be disentangling this mess, but that's out of the scope of this PR.)

Reply:
This sounds more correct, not sure why I was so fixated putting it together with the log format, moved it to json.h